main_test.go

```diff
@@ -0,0 +1,212 @@
+
+	assert.NotNil(t, db, "Database should not be nil")
+	db.Close()
+}
+
+func TestIntegrationCertCreation(t *testing.T) {
+	log.Println("Start create certificate test")
+
+	var conf Conf
+	conf.GetConf()
+
+	if conf.LogFile != "" {
+		f, err := os.OpenFile(conf.LogFile, os.O_WRONLY|os.O_CREATE, 0755)
+		if err != nil {
+			log.Fatalf("Error while reading in file: %v", err)
+		}
+		log.SetOutput(f)
+	}
+
+	if conf.Kube.SourceConfigFile == "" {
+		log.Fatal("No Kube Config file specified.")
+	}
+
+	client := buildClients(conf.Kube.SourceConfigFile)
+	testCert := createDummyCert(client, conf)
+	provisioned := checkCertStatus(client, testCert.Metadata.Name, conf)
+	assert.Equal(t, true, provisioned, "Cert has not provisioned")
+	cleanUpCert(client, testCert, conf)
+}
+
+func createDummyCert(client K8sClient, conf Conf) Certificate {
+	ranNumber := getRandomNumber()
+	name := fmt.Sprintf("integration-test-%d", ranNumber)
+	domain := fmt.Sprintf("www.integration-test-%d.com", ranNumber)
+	secret := fmt.Sprintf("integration-secret-%d", ranNumber)
+	alt := fmt.Sprintf("www.integration-alt-%d.com", ranNumber)
+
+	log.Printf("Creating test certificate - %s", name)
+	dummyCert := Certificate{
+		Metadata: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: CertificateSpec{
+			Domain:     domain,
+			Provider:   "http",
+			Email:      "integration-test@liquidweb.com",
+			SecretName: secret,
+			AltNames:   []string{alt},
+		},
+		Status: CertificateStatus{},
+	}
+	_, err := createCertificate(client.certClient, conf.Kube.NameSpace, &dummyCert)
+	if err != nil {
+		log.Fatalf("Failed to create certificate: %v", err)
+	}
+	return dummyCert
+}
+
+func checkCertStatus(client K8sClient, certName string, conf Conf) bool {
+	log.Printf("Checking certificate status of %s", certName)
+
+	var cert Certificate
+	prov := false
+	waitCtr := 0
+
+	for waitCtr < 5 || len(cert.Status.Provisioned) == 0 {
+		req := client.certClient.Get().Resource("certificates").
+			Namespace(conf.Kube.NameSpace).Name(certName)
+		err := req.Do().Into(&cert)
+
+		if err != nil {
+			log.Printf("Error while retrieving certificate: %v. Retrying", err)
+		}
+
+		time.Sleep(2 * time.Second)
+		waitCtr = waitCtr + 1
+	}
+
+	if cert.Status.Provisioned == "true" {
+		prov = true
+	}
+
+	return prov
+}
+
+func cleanUpCert(client K8sClient, testCert Certificate, conf Conf) {
+	log.Printf("Deleting test certificate %s", testCert.Metadata.Name)
+
+	req := client.certClient.Delete().Resource("certificates").
+		Namespace(conf.Kube.NameSpace).Name(testCert.Metadata.Name).Do()
+
+	if req.Error() != nil {
+		log.Printf("Error while deleting certificate: %v.", req.Error())
+	}
+
+	log.Printf("Deleting test secret %s", testCert.Spec.SecretName)
+
+	err := client.c.CoreV1().Secrets("default").Delete(testCert.Spec.SecretName, nil)
+
+	if err != nil {
+		log.Printf("Error while deleting secret: %v.", err)
+	}
+}
+
+func getRandomNumber() int {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Probably don't want to reseed the RNG each time...  maybe an Init() function?

Reply:
Sounds good